bugsnag/utils.py

```diff
@@ -118,7 +118,19 @@ def _sanitize_dict(self, obj, trim_strings, ignored):
 
         clean_dict = {}
         for key, value in six.iteritems(obj):
-            clean_dict[key] = self._sanitize(value, trim_strings, ignored)
+
+            clean_value = self._sanitize(value, trim_strings, ignored)
+
+            if isinstance(key, basestring):
+                clean_dict[key] = clean_value
+            else:
+                try:
+                    clean_dict[str(key)] = clean_value
+                except:
+                    try:
+                        clean_dict[repr(key)] = clean_value
+                    except:
+                        pass
```

Comment:
In the event an object couldn't be converted into a string using either `str` or `repr`, this could cause further problems as data would be being dropped with no indication.  If we could log the dropped data at this point, that may be a better solution.